src/v2i-hub/ERVCloudForwardingPlugin/src/ERVCloudForwardingPlugin.cpp

```diff
@@ -0,0 +1,214 @@
+#include "ERVCloudForwardingPlugin.h"
+
+namespace ERVCloudForwardingPlugin
+{
+    ERVCloudForwardingPlugin::ERVCloudForwardingPlugin(const string &name) : PluginClient(name)
+    {
+        UpdateConfigSettings();
+        std::lock_guard<mutex> lock(_cfgLock);
+        AddMessageFilter<BsmMessage>(this, &ERVCloudForwardingPlugin::handleBSM);
+        // Subscribe to all messages specified by the filters above.
+        SubscribeToMessages();
+
+        std::thread webBSM_t(&ERVCloudForwardingPlugin::StartBSMWebService, this);
+        webBSM_t.detach();
+
+        // Send RSU location to cloud
+        RegisterRSULocation();
+    }
+
+    void ERVCloudForwardingPlugin::handleBSM(BsmMessage &msg, routeable_message &routableMsg)
+    {
+        // Check if the BSM is broadcast by an ERV (Emergency Response Vehicle)
+        if (ERVCloudForwardingWorker::IsBSMFromERV(msg))
+        {
+            // Construct the ERV BSM and forward it to the cloud.
+            auto xml_str = ERVCloudForwardingWorker::constructERVBSMRequest(msg);
+            PLOG(logINFO) << "Forward ERV BSM to cloud: " << xml_str << endl;
+            CloudSendAsync(xml_str, _CLOUDURL, _CLOUDBSMREQ, _POSTMETHOD);
+        }
+        else
+        {
+            // If BSM is not from ERV, print debug log
+            PLOG(logDEBUG) << "Incoming BSM is not from Emergency Response Vehicle (ERV)." << endl;
+        }
+    }
+
+    void ERVCloudForwardingPlugin::RegisterRSULocation()
+    {
+        try
+        {
+            PLOG(logINFO) << "Create SNMP Client to connect to RSU. RSU IP:" << _rsuIp << ",\tRSU Port:" << _snmpPort << ",\tSecurity Name:" << _securityUser << ",\tAuthentication Passphrase: " << _authPassPhrase << endl;
+            auto snmpClient = std::make_shared<SNMPClient>(_rsuIp, _snmpPort, _securityUser, _authPassPhrase);
+            auto gps_sentense = snmpClient->SNMPGet(_GPSOID);
+            auto gps_map = ERVCloudForwardingWorker::ParseGPS(gps_sentense);
```

Comment:
on exception ParseGPS returns an empty MAP. Do we need to check for that here and log something?

Reply:
The caller will check if returned map has data.